### ISSUE TITLE
Inference error

### ISSUE THREAD

**Issue by joeyy5588** [da-fr/arc-prize-2024#8]
### Description
When adapting `run_evaluation_Llama-rearc_with_ttt.py` to use a Nemo model, I noticed that the tokenizer’s `model_max_length` is 8192 while the LM's `max_seq_length` remains at 2048. This mismatch leads to an error at inference time [at this line:](https://github.com/da-fr/arc-prize-2024/blob/main/training_code/inference_tools.py#L109) when input is longer than 2048, i.e., `pos` ≥ 2048, because the logits tensor only has length 2048.

### Workaround
Explicitly override `max_seq_length` when loading the model, for example in [model_tools.py:L42](https://github.com/da-fr/arc-prize-2024/blob/main/training_code/model_tools.py#L42):

```python
FastLanguageModel.from_pretrained(
    ...,
    max_seq_length=8192,
    ...
)
```

I'm wondering if this is the correct way to fix it?
Thanks for open sourcing this amazing project!

**Comment by da-fr** [comment]
Hi! Thanks for flagging this — it does appear to be a bug introduced in a newer version of `unsloth`, where `max_seq_length` is now set to a fixed value (2048) rather than using the correct value from the model’s config (accessible via `model.generation_config.max_length`).

Your workaround looks solid, and we’ve actually implemented a similar fix in the code we recently released for our ICML 2025 paper [here](https://github.com/da-fr/Product-of-Experts-ARC-Paper/blob/2800fe061e9dad25322ad6881add158468217786/model_tools.py#L64). I’ve now ported that fix back into this repository as well.

Thanks again for catching this!
